packages/unified-selection/src/unified-selection/SelectionScope.ts

```diff
@@ -227,17 +247,13 @@ async function* computeFunctionalElementSelection(
       SELECT DISTINCT ECInstanceId, ClassName FROM Element3dAncestorRelatedFunctionalElement
     `,
   ].join(" UNION ");
-  yield* executeQuery(queryExecutor, { ctes, ecsql, bindings });
+  yield* executeQuery<SelectableInstanceKey>(queryExecutor, { ctes, ecsql, bindings }, blockHandler, (row: ECSqlQueryRow) => ({
+    className: row.ClassName,
+    id: row.ECInstanceId,
+  }));
 }
 
 function formAncestorLevelBinding(ancestorLevel: number, bindings: ECSqlBinding[]) {
   bindings.push({ type: "int", value: ancestorLevel });
   return "?";
 }
-
-async function* executeQuery(queryExecutor: ECSqlQueryExecutor, query: ECSqlQueryDef): AsyncIterableIterator<SelectableInstanceKey> {
```

Comment:
At the moment all `executeQuery` calls have to pass a similar query row parser function. Maybe it makes sense to have a local `executeQuery` function that calls the more generic one with the parser? E.g.

```ts
import { executeQuery as genericExecuteQuery } from "./Utils";

async function* executeQuery(queryExecutor: ECSqlQueryExecutor, query: ECSqlQueryDef): AsyncIterableIterator<SelectableInstanceKey> {
  yield* genericExecuteQuery(queryExecutor, query, (row: ECSqlQueryRow) => ({
    className: row.ClassName,
    id: row.ECInstanceId,
  }));
}
```

Reply:
Switched to local functions.